core/src/main/java/hudson/model/Queue.java

```diff
@@ -1505,10 +1522,12 @@ public void maintain() {
 
                     // found a matching executor. use it.
                     WorkUnitContext wuc = new WorkUnitContext(p);
+                    LOGGER.log(Level.FINER, "Found a matching executor for {0}. Using it.", p.task.getFullDisplayName());
```

Comment:
Such approach causes the calculation of  p.task.getFullDisplayName() even if the message won't be printed due to the log level spec